engine/flink/kafka-util/src/main/scala/pl/touk/nussknacker/engine/flink/util/signal/KafkaSignalStreamConnector.scala

```diff
@@ -13,7 +14,11 @@ trait KafkaSignalStreamConnector {
   def connectWithSignals[A, B: TypeInformation](start: DataStream[A], processId: String, nodeId: String, schema: DeserializationSchema[B]): ConnectedStreams[A, B] = {
     val signalsSource = new FlinkKafkaConsumer[B](signalsTopic, schema,
       KafkaEspUtils.toProperties(kafkaConfig, Some(s"$processId-$nodeId-signal")))
-    val signals = start.executionEnvironment.addSource(signalsSource).name(s"signals-$processId-$nodeId")
+    val signals = start.executionEnvironment
+      .addSource(signalsSource).name(s"signals-$processId-$nodeId")
+      //We use ingestion time here, to advance watermarks in connected streams
+      //TODO: this is not always optimal solution, as e.g. in tests periodic watermarks are not the best option
+      .assignTimestampsAndWatermarks(new IngestionTimeExtractor[B])
```

Comment:
Can you extract this assigner creation to protected method? Would be nice to have ability to override it. Some examples of assigner that i would see here:
- some punctuated assigner for tests
- assigner taking timestamp from signal event or at least kafka insertion timestamp (`previousElementTimestamp`)
The default one - ingenstion time is fine for me :+1:

Reply:
Good idea, I extracted to method and added more docs